src/autopas/containers/directSum/DirectSum.h

```diff
@@ -148,6 +148,60 @@ class DirectSum : public CellBasedParticleContainer<FullParticleCell<Particle>>
                                                                           behavior, nullptr));
   }
 
+  /**
+   * @copydoc LinkedCells::forEach()
+   */
+  template <typename Lambda>
+  void forEach(Lambda forEachLambda, IteratorBehavior behavior) {
+    auto forEach = [&](ParticleCell cell) {
+      for (Particle p : cell._particles) {
+        forEachLambda(p);
+      }
+    };
+
+    std::vector<size_t> cellsOfInterest;
+
+    if (behavior & IteratorBehavior::owned) {
+      getCell().forEach(forEachLambda);
+      cellsOfInterest.push_back(0);
+    }
+    if (behavior & IteratorBehavior::halo) {
+      getHaloCell().forEach(forEachLambda);
+      cellsOfInterest.push_back(1);
+    }
+    // sanity check
+    if (cellsOfInterest.empty()) {
+      utils::ExceptionHandler::exception("Encountered invalid iterator behavior!");
+    }
```

Comment:
This could move to a private function and be reused for `reduce()` and region functions.
Also, why store the cells of interest except for the sanity check? I actually would drop that.

Reply:
check iteratorbehavior directly, wrap getCellsOfInterest